xapian-applications/omega/worker.cc

```diff
@@ -153,21 +154,36 @@ Worker::extract(const std::string& filename,
 	start_worker_subprocess();
     }
 
-    string strpage;
-
+    string strpage, strstate;
+    char state = MSG_FATAL_ERROR;
     // Sending a filename and wating for the answer
-    if (write_string(sockt, filename) &&
-	read_string(sockt, dump) &&
-	read_string(sockt, title) &&
-	read_string(sockt, keywords) &&
-	read_string(sockt, author) &&
-	read_string(sockt, strpage)) {
-	    if (strpage.empty() ||
-		strpage.find_first_not_of("0123456789") != string::npos)
-		pages = 0;
-	    else
-		pages = stoi(strpage);
-	    return true;
+    if (write_string(sockt, filename) && read_string(sockt, strstate)) {
+	if (!strstate.empty())
+	    state = strstate[0];
+	switch (state) {
+	    case MSG_OK:
+		if (!read_string(sockt, dump) ||
+		    !read_string(sockt, title) ||
+		    !read_string(sockt, keywords) ||
+		    !read_string(sockt, author) ||
+		    !read_string(sockt, strpage)) {
+		    break;
+		}
+		if (strpage.empty() ||
+		    strpage.find_first_not_of("0123456789") != string::npos) {
+		    pages = 0;
+		} else {
+		    pages = stoi(strpage);
```

Comment:
I know this wasn't added by this commit, but this ought to be fixed - it's silly to use `stoi()` but first have to check what's being parsed consists only of digits to avoid `stoi()` throwing an exception - by doing this you're essentially parsing the number twice.

Better to use the Xapian-provided `parse_unsigned()` helper function instead (see `common/parseint.h`).

Reply:
Okay, I am going to change it